docs/yaml-classes-and-relationships.rst

```diff
@@ -702,6 +708,12 @@ editable
   :Type: boolean
   :Default Value: false
   
+optional
+  :Description: If component is user_edited, and editable is true, is this an optional attribute?
```

Comment:
The idea here was that the component might have many (editable) properties, only some of which the developer would want to be in the dialog box.  The "optional" would give a potential override by narrowing the fields that would be on the form